airflow/datasets/__init__.py

```diff
@@ -50,6 +50,13 @@ def _get_uri_normalizer(scheme: str) -> Callable[[SplitResult], SplitResult] | N
     return ProvidersManager().dataset_uri_handlers.get(scheme)
 
 
+def _get_normalized_scheme(uri: str) -> str | None:
+    parsed = urllib.parse.urlsplit(uri)
+    if not parsed.scheme:
+        return None
```

Comment:
Why not just return an empty string? That’s the default behaviour of urlsplit. It’s weird you take the effort to define a new function to check the default behaviour, convert it to None, and then introduce an extra check for that new None possibility you introduced yourself.

Reply:
The reason behind a separate function was that i did not feel comfortable just re-using the `parsed.scheme.lower()`. I felt like this logic can easily be changed here and we will not be aware of it in the provider, so by creating another layer we can all use the same function. This conversion to None makes no sense and I'll remove it, but the separate function makes sense to me as we start using this normalized scheme as a key in multiple places.

Reply:
Removed returning None and adjusted the conditions everywhere.